test/expectations/expectations.go

```diff
@@ -89,6 +89,9 @@ func ExpectApplied(ctx context.Context, c client.Client, objects ...client.Objec
 	GinkgoHelper()
 	for _, o := range objects {
 		current := o.DeepCopyObject().(client.Object)
+		statuscopy := o.DeepCopyObject().(client.Object) // Snapshot the status, since create/update may override
+		deletionTimestampSet := !o.GetDeletionTimestamp().IsZero()
```

Comment:
I'm not sure that I'm convinced by the deletionTimestamp logic that's present here -- this might have been in the karpenter implementation, but I'm not sure that this makes sense and is something that we want to pull in here